src/ctap/data_formats.rs

```diff
@@ -94,16 +96,22 @@ impl From<PublicKeyCredentialUserEntity> for cbor::Value {
     }
 }
 
+// https://www.w3.org/TR/webauthn/#enumdef-publickeycredentialtype
 #[derive(Clone, PartialEq)]
 #[cfg_attr(any(test, feature = "debug_ctap"), derive(Debug))]
 pub enum PublicKeyCredentialType {
     PublicKey,
+    // This is the default for all strings not covered above.
+    // Unknown types should be ignored, instead of returning errors.
+    Unknown,
 }
 
 impl From<PublicKeyCredentialType> for cbor::Value {
     fn from(cred_type: PublicKeyCredentialType) -> Self {
         match cred_type {
             PublicKeyCredentialType::PublicKey => "public-key",
+            // We should never create this credential type.
+            PublicKeyCredentialType::Unknown => unreachable!(),
```

Comment:
Even though we indeed shouldn't create such unknown credential types, using `unreachable` here sounds like a code smell. It's only unreachable as long as no caller of this function passes the `Unknown` variant. Usually, `unreachable` can be checked locally within the current function.

So we should probably remove the `Unknown` variant of `PublicKeyCredentialType`, and either provide `impl TryFrom<&cbor::Value> for Option<PublicKeyCredentialType>`, or have another enum to layer that:

```rust
enum CredentialType {
    PublicKeyCredentialType,
    Unknown,
}
```

I'm actually surprised that the `PublicKeyCredentialType` enum contains `PublicKey` only. Shouldn't it be named `CredentialType` if it could in principle contain something else than `PublicKey`?

Reply:
Neither CTAP2 nor WebAuthn share a vision for this enum, it is simply called an "extension point". Not sure if it's worth raising a naming issue after it is already established.

Why would we need a new custom enum when `Option` solves the same problem? I see how `unreachable` is not really what we want here. To me, the next best solution would be to just return the string `"unknown"` in that case. That way, we behave similarly to the enum `SignatureAlgorithm` that assign a plain 0 to the `Unknown` variant.

The alternative is to remove `PublicKeyCredentialType` and `SignatureAlgorithm` and let `PublicKeyCredentialParameter` have 2 Options instead. The point of Unknown was to simplify, but I'll change it if you don't like it?